### PR TITLE
psp/provider_test: remove redundant tests

### LINKED ISSUE #67387: Remove wrong and redundant tests from podsecuritypolicy/provider_test.go
/kind bug

**What happened**:
In `pkg/security/podsecuritypolicy/provider_test.go`, there are tests for **{Default,}AllowPrivilegeEscalation** fields, but those in `TestValidateDefaultAllowPrivilegeEscalation()` actually test the **AllowPrivilegeEscalation** as this is set in the test instead of testing its suggested **Default\*** form, thus being mostly redundant to tests in `TestValidateAllowPrivilegeEscalation()`. On top of that, part of the tests in `TestValidateDefaultAllowPrivilegeEscalation()` test a scenario which is not allowed, that being
```
psp.Spec.DefaultAllowPrivilegeEscalation = true
psp.Spec.AllowPrivilegeEscalation = false
```
are wrong due to https://github.com/kubernetes/kubernetes/blob/fbb2dfcc6ad345b0ca3fe09cb4bc2a23ec0781d5/pkg/apis/policy/validation/validation.go#L288-L290
but since this is assigned directly, it was never validated and thus no errors were found.

**What you expected to happen**:
There are no redundant or incorrect tests in `pkg/security/podsecuritypolicy/provider_test.go`.

**Environment**:
- Kubernetes version: master

### PR DESCRIPTION
**Pull request by stlaz**
**What this PR does / why we need it**:
This PR removes redundant and incorrect tests from `pkg/security/podsecuritypolicy/provider_test.go`

**Which issue(s) this PR fixes**:
Fixes https://github.com/kubernetes/kubernetes/issues/67387

**Special notes for your reviewer**:
You can check that the test coverage is not impacted by running:
```
KUBE_COVER=y WHAT=./pkg/security/podsecuritypolicy/ make test
```

**Release note**:
```release-note
NONE
```

### CODE REVIEW

**Comment by php-coder** [comment]
PTAL @jessfraz

**Comment by php-coder** [comment]
PTAL @kubernetes/sig-auth-pr-reviews

File: pkg/security/podsecuritypolicy/provider_test.go [hunk 1]
```diff
@@ -1256,52 +1256,6 @@ func TestValidateAllowPrivilegeEscalation(t *testing.T) {
 	if len(errs) != 0 {
 		t.Errorf("directly allowing privilege escalation expected no errors but got %v", errs)
 	}
-}
-
-// TestValidateDefaultAllowPrivilegeEscalation will test that when the podSecurityPolicy
-// DefaultAllowPrivilegeEscalation is false we cannot set a container's
-// securityContext to allowPrivilegeEscalation but when it is true we can.
-func TestValidateDefaultAllowPrivilegeEscalation(t *testing.T) {
```
Comment (tallclair) [inline pkg/security/podsecuritypolicy/provider_test.go]: I'm finding these tests very hard to reason about... what do you think of this instead? https://github.com/kubernetes/kubernetes/pull/68105

Comment (liggitt) [reply pkg/security/podsecuritypolicy/provider_test.go]: I like the table test replacement

**Comment by php-coder** [comment]
@stlaz In light of #68105, is this PR and related issue can be closed?

**Comment by stlaz** [comment]
Ah, I was away for a week and was not watching my mails. Thank you @tallclair for looking into this, looks much cleaner now!
I am going to close both this PR and the issue.

**Pull request closed without merging**